autonomous/autonomous.py

```diff
@@ -137,25 +144,68 @@ def move(self, tm):
             linearVelocityRef=linear_state.velocity * 0.15,  # used for feedforward
             angleRef=goal_rotation,
         )
+        self.chassis.drive_local(
+            self.chassis_speeds.vx, self.chassis_speeds.vy, self.chassis_speeds.omega
+        )
+
         # send poses to driverstation
         display_poses = [goal_pose, self.chassis.odometry.getPose()]
         self.field.getRobotObject().setPoses(
             [trajectory_generator.goal_to_field(p) for p in display_poses]
         )
-        self.chassis.drive_local(
-            self.chassis_speeds.vx, self.chassis_speeds.vy, self.chassis_speeds.omega
-        )
-
         wpilib.SmartDashboard.putNumber("auto_vel", float(linear_state.velocity))
-        wpilib.SmartDashboard.putNumber("auto_look_around", look_around)
 
         self.last_pose = goal_pose
 
     @state
-    def stopped(self):
-        """Finished moving but still want to be trying to fire,
-        needed for second ball at terminal"""
+    def pickup(self, state_tm, tm):
+        """Waits until full"""
         self.shooter_control.wants_to_fire = True
+        self.indexer_control.wants_to_intake = True
+        if False or state_tm > 2:  # self.indexer_control.is_full():
+            self.move_next_waypoint(tm)
+            self.next_state("move")
+
+    @state(first=True)
+    def firing(self, state_tm, tm):
+        """Waits until empty"""
+        self.shooter_control.wants_to_fire = True
+        if state_tm > 1:  # TODO: replace with indexer is empty and finished firing
+            self.move_next_waypoint(tm)
+            self.next_state("move")
+
+    def move_next_waypoint(self, cur_time):
```

Comment:
Let's actually type hint all our methods and states...
```suggestion
    def move_next_waypoint(self, cur_time: float) -> None:
```
Not going to comment on every line because :zzz: